dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp

```diff
@@ -0,0 +1,974 @@
+        {
+            if (!m_isPaused)
+            {
+                m_controller.Pause();
+            }
+        }
+    }
+}
+
+// Called when the animation was hidden but is now becoming visible.
+void AnimatedVisualPlayer::AnimationPlay::OnUnhiding()
+{
+    if (m_isPausedBecauseHidden)
+    {
+        m_isPausedBecauseHidden = false;
+
+        // Resume the animation that was paused due to the app being suspended.
+        if (m_controller)
+        {
+            if (!m_isPaused)
+            {
+                m_controller.Resume();
+            }
+        }
+    }
+}
+
+void AnimatedVisualPlayer::AnimationPlay::Pause()
+{
+    m_isPaused = true;
+
+    if (m_controller)
+    {
+        if (!m_isPausedBecauseHidden)
+        {
+            m_controller.Pause();
+        }
+    }
+}
+
+void AnimatedVisualPlayer::AnimationPlay::Resume()
+{
+    m_isPaused = false;
+
+    if (m_controller)
+    {
+        if (!m_isPausedBecauseHidden)
+        {
+            m_controller.Resume();
+        }
+    }
+}
+
+// Completes the play, and unregisters it from the player.
+// Do not do anything with this object after calling Complete()... the object is destructed already.
+void AnimatedVisualPlayer::AnimationPlay::Complete()
+{
+    //
+    // NOTEs about lifetime (i.e. why we can trust that m_owner is still valid)
+    //  The AnimatedVisualPlayer will always outlive the AnimationPlay. This
+    //  is because:
+    //  1. There is only ever one un-completed AnimationPlay. When a new play
+    //     is started the current play is completed.
+    //  2. An uncompleted AnimationPlay will be completed when the AnimatedVisualPlayer
+    //     is unloaded.
+    //  3. Completion as a result of a call to SetProgress is always synchronous and is
+    //     called from the AnimatedVisualPlayer.
+    //  4. If the batch completion event fires, the AnimatedVisualPlayer must still
+    //     alive because if it had been unloaded Complete() would have been called
+    //     during the unload which would have unsubscribed from the batch completion
+    //     event.
+    //    
+
+    // Grab a copy of the pointer so the object stays alive until
+    // the method returns.
+    auto me = m_owner.m_nowPlaying;
+
+    // Unsubscribe from batch.Completed.
+    if (m_batch)
+    {
+        m_batch.Completed(m_batchCompletedToken);
+        m_batchCompletedToken = { 0 };
+    }
+
+    // If this play is the one that is currently associated with the player,
+    // disassociate it from the player and update the player's IsPlaying property.
+    if (IsCurrentPlay())
+    {
+        // Disconnect from the player.
+        m_owner.m_nowPlaying.reset();
+
+        // Update the IsPlaying state. Note that this is done
+        // after disconnected so that we won't be reentered.
+        m_owner.IsPlaying(false);
+    }
+
+    // Allow the play to complete.
+    CompleteAwaits();
+}
+
+AnimatedVisualPlayer::AnimatedVisualPlayer()
+{
+    EnsureProperties();
```

Comment:
Add this macro:
__RP_Marker_ClassById(RuntimeProfiler::ProfId_AnimatedVisualPlayer);

In order to instrument class usage.  And include "RuntimeProfiler.h" to pull in the macro definition.